### PR TITLE
UAF-3794 :  fix Exception when calculate PREQ with 0 total amount

### PR DESCRIPTION
**Pull request by lshyu**
Based on clarification, generated distribution accounts from first line item accounts if PREQ total amount is 0.

### CODE REVIEW

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/PaymentRequestServiceImpl.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.commons.lang.StringUtils;
 import org.kuali.kfs.module.purap.PurapConstants;
+import org.kuali.kfs.module.purap.businessobject.CreditMemoAccount;
```
Comment (sskinner) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/PaymentRequestServiceImpl.java]: Dead import, please remove.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                }
+            }
+        }
+        // iterate over useTaxAccounts and set summary accounts using proration
+        for (UseTaxContainer useTaxContainer : useTaxAccounts) {
+
+            // create summary from items
+            List<SourceAccountingLine> origSourceAccounts = this.generateSummaryWithNoZeroTotals(useTaxContainer.getItems());
+            KualiDecimal totalAmount = calculateSumTotal(origSourceAccounts);
+            // UAF-3794 only this statement changed
```
Comment (sskinner) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java:128]: Thank you calling out the diff from super's method.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    	
+        for (PurApItem item : purapItems) {
+            if (item.getItemType().isLineItemIndicator()) {
+            	for (PurApAccountingLine account : item.getSourceAccountingLines()) {
+                    PurApAccountingLine newAccountingLine = null;
+                    try {
+                        newAccountingLine = (PurApAccountingLine) clazz.newInstance();
+                    }
+                    catch (InstantiationException e) {
+                        e.printStackTrace();
```
Comment (sskinner) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java]: I'm not a fan of printing STEs directly to STDOUT -- I'd suggest either log out at an appropriate level, or swallow the exception entirely with a comment explaining why that's not a bad thing (e.g.: // ObjectPopulationUtils can handle a this nulled out reference... I have no idea if that's the case).

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                    }
+                    catch (IllegalAccessException e) {
+                        e.printStackTrace();
```
Comment (sskinner) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java]: Another log/comment opportunity.

**Review by sskinner — CHANGES_REQUESTED** [review]
This all looks really solid, the only thing sparking a change request, are the catch blocks throwing STEs to stdout. And if we're addressing those, might as well pull out the dead import. Overall really great job!

**Comment by lshyu** [comment]
Scott,

   I made changes based on your comment.  Throwing STE to stdout was a copy from 'org'.  Please let me know if there is further change.  Thanks.

**Review by sskinner — APPROVED** [review]
Nice and succinct to fail fast _if_ that account object can't be instantiated; I like it. Thanks for these changes.

**Pull request merged**